lib/Conversion/TorchOnnxToTorch/OnnxLstmExpander.cpp

```diff
@@ -0,0 +1,479 @@
+#include "torch-mlir/Conversion/TorchOnnxToTorch/OnnxLstmExpander.h"
+#include "torch-mlir/Conversion/TorchOnnxToTorch/Patterns.h"
+#include "torch-mlir/Conversion/TorchOnnxToTorch/Utils.h"
+#include "torch-mlir/Dialect/Torch/IR/TorchOps.h"
+#include "torch-mlir/Dialect/Torch/Utils/Utils.h"
+
+namespace mlir::torch::onnx_c {
+
+Value createActivationByName(ConversionPatternRewriter &rewriter, Location loc,
+                             StringRef name, Value input) {
+  if (name == "Sigmoid") {
+    return rewriter.create<Torch::AtenSigmoidOp>(loc, input.getType(), input);
+  } else if (name == "Tanh") {
+    return rewriter.create<Torch::AtenTanhOp>(loc, input.getType(), input);
+  } else if (name == "Relu") {
+    return rewriter.create<Torch::AtenReluOp>(loc, input.getType(), input);
+  } else {
+    llvm_unreachable("Unsupported activation function");
+  }
+}
+
+std::pair<Value, Value>
+lstm_cell(OpBinder binder, ConversionPatternRewriter &rewriter,
+          Value Xt,     // =input_seq shape [batch_size, input_size]
+          Value H_prev, // =hidden_state shape [batch_size, hidden_size]
+          Value C_prev, // =cell_state shape [batch_size, hidden_size]
+          Value W_i, Value W_o, Value W_f, Value W_c,
+          // W shape [hidden_size, input_size]
+          Value Wb_i, Value Wb_o, Value Wb_f, Value Wb_c,
+          // Wb shape [hidden_size]
+          Value R_i, Value R_o, Value R_f, Value R_c,
+          // R shape [hidden_size, hidden_size]
+          Value Rb_i, Value Rb_o, Value Rb_f, Value Rb_c,
+          // Rb shape [hidden_size]
+          SmallVector<std::string> activations) {
+  Location loc = binder.getLoc();
+  auto intType = rewriter.getType<Torch::IntType>();
+  Torch::ValueTensorType HType =
+      H_prev.getType().cast<Torch::ValueTensorType>();
+
```

Comment:
You can reduce some of the size by using an `ImplicitLocOpBuilder`. It allows you to set the location information once:
```
ImplicitLocOpBuilder b(loc, rewriter);
b.create<OpTy>(opArgs);
```

It removes the location information requirement from the builder.